### PR TITLE
Add hook support for tracking changes when updating meta by mid

### LINKED ISSUE #5903: Support `metadata_by_mid` filters
Currently metadata functions that use the metadata ID do not work with Pods.

- `update_metadata_by_mid`
- `get_metadata_by_mid`
- `delete_metadata_by_mid`

Report: https://wordpress.org/support/topic/edit-woocommerce-product-with-relationship-field-with-api-rest/#post-13838071

### PR DESCRIPTION
**Pull request by sc0ttkclark**
Fixes #5903

## Description

Adds hook support for tracking changes when updating meta by mid through the `update_metadata_by_mid ()` core function

## Changelog text for these changes

Added: New support for tracking field changes when updating meta records manually through the `update_metadata_by_mid ()` core function which integrates with WooCommerce REST API meta saving for products.

## PR Checklist

- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.
- [ ] My code includes automated tests for PHP and/or JS (if applicable).

### CODE REVIEW

File: classes/PodsMeta.php [hunk 1]
```diff
@@ -151,6 +151,7 @@ public function core() {
 			if ( ! pods_tableless() ) {
 				add_filter( 'add_post_metadata', array( $this, 'add_post_meta' ), 10, 5 );
 				add_filter( 'update_post_metadata', array( $this, 'update_post_meta' ), 10, 5 );
+				add_filter( 'update_post_metadata_by_id', array( $this, 'update_post_meta_by_id' ), 10, 4 );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:154]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 94959fd1ca1a9e3172043d2c40919cff</sub>

File: classes/PodsMeta.php [hunk 2]
```diff
@@ -193,6 +194,7 @@ public function core() {
 					if ( ! pods_tableless() ) {
 						add_filter( 'add_term_metadata', array( $this, 'add_term_meta' ), 10, 5 );
 						add_filter( 'update_term_metadata', array( $this, 'update_term_meta' ), 10, 5 );
+						add_filter( 'update_term_metadata_by_id', array( $this, 'update_term_meta_by_id' ), 10, 4 );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:197]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: c79682639f6e8310406e7c84652ce115</sub>

File: classes/PodsMeta.php [hunk 3]
```diff
@@ -234,6 +236,7 @@ public function core() {
 					if ( ! pods_tableless() ) {
 						add_filter( 'add_post_metadata', array( $this, 'add_post_meta' ), 10, 5 );
 						add_filter( 'update_post_metadata', array( $this, 'update_post_meta' ), 10, 5 );
+						add_filter( 'update_post_metadata_by_id', array( $this, 'update_post_meta_by_id' ), 10, 4 );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:239]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 770ede8cdb14e9928b91a738b78da094</sub>

File: classes/PodsMeta.php [hunk 4]
```diff
@@ -260,6 +263,7 @@ public function core() {
 				if ( ! pods_tableless() ) {
 					add_filter( 'add_user_metadata', array( $this, 'add_user_meta' ), 10, 5 );
 					add_filter( 'update_user_metadata', array( $this, 'update_user_meta' ), 10, 5 );
+					add_filter( 'update_user_metadata_by_id', array( $this, 'update_user_meta_by_id' ), 10, 4 );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:266]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 8c478ab2b7791d39f2ee04af0936c7df</sub>

File: classes/PodsMeta.php [hunk 5]
```diff
@@ -285,6 +289,7 @@ public function core() {
 				if ( ! pods_tableless() ) {
 					add_filter( 'add_comment_metadata', array( $this, 'add_comment_meta' ), 10, 5 );
 					add_filter( 'update_comment_metadata', array( $this, 'update_comment_meta' ), 10, 5 );
+					add_filter( 'update_comment_metadata_by_id', array( $this, 'update_comment_meta_by_id' ), 10, 4 );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:292]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 1816561c9c397abe7e73758a436b49ba</sub>

File: classes/PodsMeta.php [hunk 6]
```diff
@@ -2858,6 +2863,126 @@ public function update_option() {
...
+		 * @param null|bool $_null Whether to override the meta handling by Pods.
+		 * @param array     $args  The function arguments with the type added to the front.
+		 */
+		$_null = apply_filters( 'pods_meta_update_post_meta_by_id', null, $args );
+
+		if ( null !== $_null ) {
+			return $_null;
+		}
+
+		return call_user_func_array( array( $this, 'update_meta_by_id' ), $args );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:2893]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 5576a3e5c59c7200e6c9acbda373551d</sub>

File: classes/PodsMeta.php [hunk 7]
```diff
@@ -2858,6 +2863,126 @@ public function update_option() {
...
+		 * @param null|bool $_null Whether to override the meta handling by Pods.
+		 * @param array     $args  The function arguments with the type added to the front.
+		 */
+		$_null = apply_filters( 'pods_meta_update_user_meta_by_id', null, $args );
+
+		if ( null !== $_null ) {
+			return $_null;
+		}
+
+		return call_user_func_array( array( $this, 'update_meta_by_id' ), $args );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:2923]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 97e715febeff2562db27e98f99cbd2b1</sub>

File: classes/PodsMeta.php [hunk 8]
```diff
@@ -2858,6 +2863,126 @@ public function update_option() {
...
+		 * @param null|bool $_null Whether to override the meta handling by Pods.
+		 * @param array     $args  The function arguments with the type added to the front.
+		 */
+		$_null = apply_filters( 'pods_meta_update_comment_meta_by_id', null, $args );
+
+		if ( null !== $_null ) {
+			return $_null;
+		}
+
+		return call_user_func_array( array( $this, 'update_meta_by_id' ), $args );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:2953]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 80b90af61f2933fd30c56165c9ccceb4</sub>

File: classes/PodsMeta.php [hunk 9]
```diff
@@ -2858,6 +2863,126 @@ public function update_option() {
...
+		 * @param null|bool $_null Whether to override the meta handling by Pods.
+		 * @param array     $args  The function arguments with the type added to the front.
+		 */
+		$_null = apply_filters( 'pods_meta_update_term_meta_by_id', null, $args );
+
+		if ( null !== $_null ) {
+			return $_null;
+		}
+
+		return call_user_func_array( array( $this, 'update_meta_by_id' ), $args );
```
Comment (tr1b0t) [inline classes/PodsMeta.php:2983]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: f7e62fb27e6df3b1303ae98fcf491242</sub>

File: includes/general.php [hunk 10]
```diff
@@ -2346,6 +2346,7 @@ function pods_no_conflict_on( $object_type = 'post', $object = null ) {
 				$no_conflict['filter'] = array_merge( $no_conflict['filter'], array(
 					array( 'add_post_metadata', array( PodsInit::$meta, 'add_post_meta' ), 10, 5 ),
 					array( 'update_post_metadata', array( PodsInit::$meta, 'update_post_meta' ), 10, 5 ),
+					array( 'update_post_metadata_by_id', array( PodsInit::$meta, 'update_post_meta_by_id' ), 10, 4 ),
```
Comment (tr1b0t) [inline includes/general.php:2349]: * :no_entry_sign: Short array syntax must be used to define arrays
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: a1b26ff26ed75a4ec5ec67ecac0a65a8</sub>

File: includes/general.php [hunk 11]
```diff
@@ -2371,6 +2372,7 @@ function pods_no_conflict_on( $object_type = 'post', $object = null ) {
 				$no_conflict['filter'] = array_merge( $no_conflict['filter'], array(
 					array( 'add_term_metadata', array( PodsInit::$meta, 'add_term_meta' ), 10, 5 ),
 					array( 'update_term_metadata', array( PodsInit::$meta, 'update_term_meta' ), 10, 5 ),
+					array( 'update_term_metadata_by_id', array( PodsInit::$meta, 'update_term_meta_by_id' ), 10, 4 ),
```
Comment (tr1b0t) [inline includes/general.php:2375]: * :no_entry_sign: Short array syntax must be used to define arrays
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: f9596bf81f8eccf48222c6dc555a2a48</sub>

File: includes/general.php [hunk 12]
```diff
@@ -2402,6 +2404,7 @@ function pods_no_conflict_on( $object_type = 'post', $object = null ) {
 				$no_conflict['filter'] = array_merge( $no_conflict['filter'], array(
 					array( 'add_post_metadata', array( PodsInit::$meta, 'add_post_meta' ), 10, 5 ),
 					array( 'update_post_metadata', array( PodsInit::$meta, 'update_post_meta' ), 10, 5 ),
+					array( 'update_post_metadata_by_id', array( PodsInit::$meta, 'update_post_meta_by_id' ), 10, 4 ),
```
Comment (tr1b0t) [inline includes/general.php:2407]: * :no_entry_sign: Short array syntax must be used to define arrays
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 4952f4347b446f7b170387dbb474210e</sub>

File: includes/general.php [hunk 13]
```diff
@@ -2423,6 +2426,7 @@ function pods_no_conflict_on( $object_type = 'post', $object = null ) {
 				$no_conflict['filter'] = array_merge( $no_conflict['filter'], array(
 					array( 'add_user_metadata', array( PodsInit::$meta, 'add_user_meta' ), 10, 5 ),
 					array( 'update_user_metadata', array( PodsInit::$meta, 'update_user_meta' ), 10, 5 ),
+					array( 'update_user_metadata_by_id', array( PodsInit::$meta, 'update_user_meta_by_id' ), 10, 4 ),
```
Comment (tr1b0t) [inline includes/general.php:2429]: * :no_entry_sign: Short array syntax must be used to define arrays
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 29fece6c39e82f32d02ee8d9a480312b</sub>

File: includes/general.php [hunk 14]
```diff
@@ -2448,6 +2452,7 @@ function pods_no_conflict_on( $object_type = 'post', $object = null ) {
 				$no_conflict['filter'] = array_merge( $no_conflict['filter'], array(
 					array( 'add_comment_metadata', array( PodsInit::$meta, 'add_comment_meta' ), 10, 5 ),
 					array( 'update_comment_metadata', array( PodsInit::$meta, 'update_comment_meta' ), 10, 5 ),
+					array( 'update_comment_metadata_by_id', array( PodsInit::$meta, 'update_comment_meta_by_id' ), 10, 4 ),
```
Comment (tr1b0t) [inline includes/general.php:2455]: * :no_entry_sign: Short array syntax must be used to define arrays
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 8abcaa39a6b73360b20438ce9230803e</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 19 error(s)

File: classes/PodsMeta.php (hunk 6 above)
Comment (sc0ttkclark) [reply classes/PodsMeta.php:2893]:
```suggestion
		return call_user_func_array( [ $this, 'update_meta_by_id' ], $args );
```

File: classes/PodsMeta.php (hunk 7 above)
Comment (sc0ttkclark) [reply classes/PodsMeta.php:2923]:
```suggestion
		return call_user_func_array( [ $this, 'update_meta_by_id' ], $args );
```

File: classes/PodsMeta.php (hunk 8 above)
Comment (sc0ttkclark) [reply classes/PodsMeta.php:2953]:
```suggestion
		return call_user_func_array( [ $this, 'update_meta_by_id' ], $args );
```

File: classes/PodsMeta.php (hunk 9 above)
Comment (sc0ttkclark) [reply classes/PodsMeta.php:2983]:
```suggestion
		return call_user_func_array( [ $this, 'update_meta_by_id' ], $args );
```

**Pull request closed without merging**